druntime/src/core/thread/osthread.d

```diff
@@ -1819,12 +1819,17 @@ extern (C) void thread_suspendAll() nothrow
         Thread.criticalRegionLock.lock_nothrow();
         scope (exit) Thread.criticalRegionLock.unlock_nothrow();
         size_t cnt;
+        bool suspendedSelf;
         Thread t = ThreadBase.sm_tbeg.toThread;
         while (t)
         {
             auto tn = t.next.toThread;
             if (suspend(t))
+            {
+                if (t is ThreadBase.getThis())
```

Comment:
Not that the performance is critical here, but maybe store `ThreadBase.getThis` in a var instead of calling it every time.

Reply:
Druntime is compiled all at once, so the hope is that it'll be inlined. ;-)

Reply:
Ah yeah, I was thinking it was a virtual call. Duh...

Reply:
Also, see the NOTE at the top of this function.